src/App.js

```diff
@@ -2,20 +2,53 @@ import React from 'react';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
-const useSemiPersistentState = () => {
-  const [todoList, setTodoList] = React.useState(JSON.parse(localStorage.getItem("savedTodoList")) || [])
+
+
+function App() {
   
-  React.useEffect(() => {
-    console.log('todoList has changed:', todoList);
-    localStorage.setItem("savedTodoList", JSON.stringify(todoList));
-  }, [todoList]);
   
-  return [todoList, setTodoList]
-}
+  const [todoList, setTodoList] = React.useState([])
 
-function App() {
-  const [todoList, setTodoList] = useSemiPersistentState();
+  const [isLoading, setIsLoading] = React.useState(true);
+
+
+  React.useEffect(() => {
+    const myPromise = new Promise((resolve, reject) => {
+      setTimeout(() => {
+        const result = {
+          data: { todoList: JSON.parse(localStorage.getItem("savedTodoList") || []) }
+        };
+
+        resolve(result);
+      }, 2000);
+              
+    }, [])
+    
+    myPromise
+      .then(result => {
+        setTodoList(result.data.todoList);
+        setIsLoading(false);
+      })
+      .catch(error => {
+        console.error(error);
+        
+      });
+    
+  }); 
```

Comment:
The `useEffect` hook is missing its dependency array. I would remove the `[]` from line 25